### PR TITLE
Feature risk vs dist popup

### PR DESCRIPTION
**Pull request by Zach10a**
When a path is right-clicked in the layer menu you can now see the following information

image

### CODE REVIEW

File: seedpod_ground_risk/core/plot_server.py [hunk 1]
```diff
@@ -329,6 +330,16 @@ def export_path_geojson(self, layer, filepath):
         if layer in self.annotation_layers:
             layer.dataframe.to_file(os.path.join(os.sep, f'{filepath}', 'path.geojson'), driver='GeoJSON')
 
+    def generate_path_data_popup(self, layer):
+        from seedpod_ground_risk.pathfinding.environment import GridEnvironment
+        from seedpod_ground_risk.ui_resources.info_popups import DataWindow
+        if layer in self.annotation_layers:
+            path = layer.path
+            cur_layer = GridEnvironment(self._generated_data_layers['Fatality Risk'][1])
```
Comment (aliaksei135) [inline seedpod_ground_risk/core/plot_server.py:338]: How do you know that the layer has been generated and exists there?

File: seedpod_ground_risk/main.py [hunk 2]
```diff
@@ -103,6 +106,7 @@ def mousePressEvent(self, event: PySide2.QtGui.QMouseEvent) -> None:
             menu.addAction("Edit Layer", self.edit_layer)
             if isinstance(self._layer, AnnotationLayer):
                 menu.addAction("Export .GeoJSON", self.export_path_json)
+                menu.addAction("Show path data", self.path_data)
```
Comment (aliaksei135) [inline seedpod_ground_risk/main.py:109]: Show path data isn't a valid action for all `AnnotationLayer`s, there are other classes that inherit `AnnotationLayer` that wouldn't know how to/couldn't handle this action

File: seedpod_ground_risk/main.py [hunk 3]
```diff
@@ -243,10 +247,13 @@ def layer_reorder(self):
...
-        wizard = AircraftWizard(self, Qt.Window)
-        wizard.exec()
-        ac = wizard.d
-        add_aircraft(ac)
+        try:
+            wizard = AircraftWizard(self, Qt.Window)
+            wizard.exec()
+            ac = wizard.d
+            add_aircraft(ac)
+        except:
```
Comment (aliaksei135) [inline seedpod_ground_risk/main.py:255]: What exception does this throw? Can you handle only that exception on it's own? This kind of construct is just begging for a hidden bug somewhere

File: seedpod_ground_risk/ui_resources/info_popups.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import matplotlib.pyplot as mpl
+from PySide2.QtWidgets import QDialog, QVBoxLayout
+from matplotlib import gridspec
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
+from matplotlib.backends.backend_qt5agg import NavigationToolbar2QT as NavigationToolbar
+
+from seedpod_ground_risk.pathfinding.moo_ga import *
```
Comment (aliaksei135) [inline seedpod_ground_risk/ui_resources/info_popups.py:7]: Why is this imported?

File: seedpod_ground_risk/ui_resources/info_popups.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        ax2.text(0.5, 0.5, data_txt,
+                 horizontalalignment='center',
+                 verticalalignment='center',
+                 fontsize=10, color='black',
+                 transform=ax2.transAxes, wrap=True)
+
+        self.canvas.draw()
+        self.show()
+
+    def total_risk(self, ys):
```
Comment (aliaksei135) [inline seedpod_ground_risk/ui_resources/info_popups.py:63]: This 4 functions are effectively doing the same thing. You could replace them with a higher order function:
``` python
def op_format(self, val, op):
    return "{:.2e}".format(op(val))
```
And call it with a function reference like this:
```  python
op_format(ys, np.average)
```

File: seedpod_ground_risk/ui_resources/info_popups.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        dist = []
+        for idx in range(len(path[:-1])):
+            n0 = self.path[idx].position
+            n1 = self.path[idx + 1].position
+            l = line(n0[0], n0[1], n1[0], n1[1])
+            ys.append(grid[l[0], l[1]])
+        ys = np.hstack(ys)
+        ax1.plot(ys)
+        ax1.set_xlabel('Distance (m)')
+        ax1.set_ylabel('Risk of fatality (per hour)')
```
Comment (aliaksei135) [inline seedpod_ground_risk/ui_resources/info_popups.py:44]: Units go in square brackets eg `[m]`

**Review by aliaksei135 — APPROVED** [review]

File: seedpod_ground_risk/ui_resources/info_popups.py (hunk 6 above)
Comment (Zach10a) [reply seedpod_ground_risk/ui_resources/info_popups.py:44]: Done

File: seedpod_ground_risk/ui_resources/info_popups.py (hunk 5 above)
Comment (Zach10a) [reply seedpod_ground_risk/ui_resources/info_popups.py:63]: Done

File: seedpod_ground_risk/ui_resources/info_popups.py (hunk 4 above)
Comment (Zach10a) [reply seedpod_ground_risk/ui_resources/info_popups.py:7]: Imported to allow for numpy and 'line' function to be used.

**Comment by Zach10a** [comment]
All sorted, final feature before v0.14.0

**Review by aliaksei135 — APPROVED** [review]
Cracking Ta!

**Pull request merged**